Sources/FoundationEssentials/URL/URL.swift

```diff
@@ -1349,31 +1349,36 @@ public struct URL: Equatable, Sendable, Hashable {
         }
     }
 
-    private static func windowsPath(for posixPath: String) -> String {
-        let utf8 = posixPath.utf8
-        guard utf8.count >= 4 else {
-            return posixPath
+    private static func windowsPath(for urlPath: String) -> String {
+        let utf8 = urlPath.utf8
+        guard !utf8.starts(with: [._slash, ._slash]) else {
```

Comment:
What about `\\server\share\directory\`? Should that not be `\\server\share\directory`?

Reply:
Ideally, yeah. I think it gets a bit complicated to make sure we do the right thing for all paths, though. I'm not super familiar with UNC paths, so please correct me if I'm wrong, but I think we could have:

```
\\.\c:\
\\.\BootPartition\
\\server\share\
\\.\server\share\
\\.\UNC\server\share\
```

so we couldn't rely on just stripping after X segments. And it still might be complex to check for `.` or `?` in the first segment since we could have either `driveName` or `server\share` following it.

Do you think we should call `PathIsRootW` to determine whether we should strip the slash? I'm a bit hesitant to do filesystem I/O in `URL.path`, but I think that would get us the correct answer.

Reply:
Yeah, `PathIsRootW` is the right thing to use IMO. `\\.\` paths are device paths, there are also kernel paths `\\?\`. Additionally, you might also encounter a root local path with a `\??\` prefix. For the UNC path, you can also label it via link, e.g. `\\?\UNC\server\share\directory\`.

Reply:
Switched to checking `PathIsRootW` for non-drive letter paths in [9d325ee](https://github.com/swiftlang/swift-foundation/pull/1038/commits/9d325ee65d411131f89a1ce34349c53e007efd52)